app/controllers/application_controller.rb

```diff
@@ -137,4 +129,12 @@ def set_locale
   def sp_session
     session.fetch(:sp, {})
   end
+
+  def create_branded_experience
+    return unless session[:sp]
+
+    @sp_logo = current_sp.logo
```

Comment:
Yep. Great minds thing alike. I was going to create a view object in a separate PR.